django_project/base/views/custom_domain.py

```diff
@@ -155,3 +160,54 @@ def get_queryset(self, queryset=None):
             return queryset
 
         return self.queryset
+
+
+class DomainDeleteView(StaffuserRequiredMixin, DomainMixin, DeleteView):
+    context_object_name = 'domain'
+    template_name = 'custom_domain/delete.html'
+
+    def get_success_url(self):
+        return reverse('domain-list')
+
+    def get_queryset(self):
+        qs = Domain.objects.all()
+        if self.request.user.is_staff:
+            return qs
+        else:
+            raise Http404
+
+
+class DomainUpdateView(LoginRequiredMixin, DomainMixin, UpdateView):
+    context_object_name = 'domain'
+    template_name = 'custom_domain/update.html'
+
+    def get_form_kwargs(self):
+        kwargs = super(DomainUpdateView, self).get_form_kwargs()
+        kwargs.update({'user': self.request.user})
+        return kwargs
+
+    def get_queryset(self):
+        qs = Domain.objects
+        if self.request.user.is_staff:
+            return qs
+        else:
+            return qs.filter(user=self.request.user)
+
+    def get_context_data(self, **kwargs):
+        context = super(DomainUpdateView, self).get_context_data(**kwargs)
+        return context
+
+    def get_success_url(self):
+        if self.object.approved:
+            return reverse('domain-list')
+        else:
+            return reverse('domain-pending-list')
+
+    def form_valid(self, form):
```

Comment:
I would remove this block.
According to the model `Domain`, there is no reason why you would have a `IntegrityError` so the `try...except` is hiding us a real problem with the DB. According to the models.py, you can have two domains with the same name.

Reply:
Hmm yes, it was just for precaution I think... But maybe you are right, it is not necessary here.

No, the domain name is unique so there should not be two domains with the same name (ref https://github.com/kartoza/projecta/blob/develop/django_project/base/models/custom_domain.py#L35)

Reply:
Oups sorry, I was not looking at good line.
I try to be careful when we catch too broad exceptions. It can hide a real problem with the DB and we raise another error to the user.